hubblestack_nova/stat.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+import logging
+
+import fnmatch
+import yaml
+import os
+import copy
+import salt.utils
+
+from distutils.version import LooseVersion
+
+log = logging.getLogger(__name__)
+
+__tags__ = None
+__data__ = None
+
+
+def __virtual__():
+    if salt.utils.is_windows():
+        return False, 'This audit module only runs on linux'
+    global __tags__
+    global __data__
+    yamldir = os.path.dirname(__file__)
+    if not yamldir:
+        yamldir = '.'
+    __data__ = _get_yaml(yamldir)
+    __tags__ = _get_tags(__data__)
+    return True
+
+
+def audit(tags, verbose=False):
+    '''
+    Run the grep audits contained in the YAML files processed by __virtual__
+    '''
+    ret = {'Success': [], 'Failure': []}
+
+    for tag in __tags__:
+        if fnmatch.fnmatch(tag, tags):
+            for tag_data in __tags__[tag]:
+                name = tag_data['name']
+                expected = {}
+                for e in ['mode', 'user', 'group']:
+                    if e in tag_data:
+                        expected[e] = str(tag_data[e])
+
+                #getting the stats using salt
+                salt_ret = __salt__['file.stats'](name)
+                if not salt_ret:
+                    if 'cannot stat' in expected.values():
```

Comment:
I'm thinking instead of `'cannot stat'` maybe we should use the python value `None`.

`if None in expected.values():`

I don't like using arbitrary strings like this, even if `cannot stat` is not a valid username.